src/info/utils/git.rs

```diff
@@ -78,8 +83,27 @@ impl Commits {
             }
         }
 
-        let mut authors_by_number_of_commits: Vec<(Sig, usize)> =
-            author_to_number_of_commits.into_iter().collect();
+        let mut authors_by_number_of_commits: Vec<(Sig, usize)> = Vec::new();
+        for (e, name_to_nbr_of_commits) in email_to_names.into_iter() {
+
+            let mut most_frequent_name: Option<BString> = None;
+            let mut highest_nbr_of_commits = 0;
+            let mut author_nbr_of_commits = 0;
+
+            for (n, c) in name_to_nbr_of_commits.into_iter() {
+
+                author_nbr_of_commits += c;
+
+                if c > highest_nbr_of_commits {
+                    highest_nbr_of_commits = c;
+                    most_frequent_name.insert(n.clone());
+                }
+            }
+
+            let sig = Sig { email: e, name: most_frequent_name.unwrap().clone() };
+            authors_by_number_of_commits.push((sig, author_nbr_of_commits));
+        }
```

Comment:
This is fine with me, but I think it's a bit more idiomatic to write something more functional, for example using `Iterator::fold` or `Iterator::max_by`, and also `Iterator::collect`.
```rust
let authors_by_number_of_commits = email_to_names.into_iter().map(|(email, name_to_commits)| {
    let most_frequent_name = name_to_commits.into_iter().max_by(|(name, commits) {
        // implementation
        // NOTE: You may need to use `.fold` instead to reduce to one iteration.
    });
    Sig { /* ... */ }; // Now this is an iterator of Sigs that can be collected
}).collect();
```
IIRC the Rust compiler is pretty good about unfolding iterators into something more efficient, though I may be wrong.

I haven't used Rust extensively in a while (should I say I've gotten rusty? :laughing:) so some of this may be incorrect.